### PR TITLE
feat: detailed error handling

### PR DESCRIPTION
**Pull request by setchy**
Improve error handling feedback loop by inspecting err.response and determining root cause of error.

Refactored the previous `Oops` view to allow error types to be passed in.

| Error Type  | Example |
| ------------- | ------------- |
| Bad credentials  | Screenshot 2024-04-06 at 6 33 48 PM |
| Missing scopes |Screenshot 2024-04-06 at 6 34 07 PM |
| Rate limited |Screenshot 2024-04-06 at 6 35 31 PM |
| Default case (existing) | Screenshot 2024-04-06 at 6 35 47 PM|

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -1,2 +1,2 @@
 pnpx lint-staged
-pnpm test -- --onlyChanged
\ No newline at end of file
+pnpm test -- --onlyChanged
```
Comment (bmulholland) [inline .husky/pre-commit:2]: And back again? Permissions too.

Either way is fine, but probably best to keep it stable

File: src/utils/helpers.ts [hunk 2]
```diff
@@ -78,11 +78,14 @@ export function formatSearchQueryString(
...
-  ).data;
-
-  return response.html_url;
+  try {
+    const response: Issue | IssueComments | PullRequest = (
+      await apiRequestAuth(url, 'GET', token)
+    ).data;
+    return response.html_url;
+  } catch (err) {
+    console.error('Failed to get html url');
```
Comment (bmulholland) [inline src/utils/helpers.ts:87]: Confirming this is intentionally still there?

Comment (setchy) [reply src/utils/helpers.ts:87]: Yes, intentional for now.

File: src/utils/subject.ts [hunk 3]
```diff
@@ -137,49 +137,57 @@ async function getGitifySubjectForIssue(
...
+      state: prState,
+      user: {
+        login: prCommentUser?.login ?? pr.user.login,
+        html_url: prCommentUser?.html_url ?? pr.user.html_url,
+        avatar_url: prCommentUser?.avatar_url ?? pr.user.avatar_url,
+        type: prCommentUser?.type ?? pr.user.type,
+      },
+    };
+  } catch (err) {
+    console.error('Pull Request subject retrieval failed');
```
Comment (bmulholland) [inline src/utils/subject.ts:189]: Intended?

File: src/utils/subject.ts [hunk 4]
```diff
@@ -250,11 +258,19 @@ async function getLatestCommentUser(
...
+        'GET',
+        token,
+      )
+    )?.data;
+
+    return (
+      (response as IssueComments)?.user ?? (response as ReleaseComments).author
+    );
+  } catch (err) {
+    console.error('Discussion latest comment retrieval failed');
```
Comment (bmulholland) [inline src/utils/subject.ts:274]: Ditto

**Review by bmulholland — APPROVED** [review]
Great improvement, very user friendly. Really love this architecture, with the data structure. Clean and clear, easy to modify later. Nice work!

**Comment by setchy** [comment]
> Great improvement, very user friendly. Really love this architecture, with the data structure. Clean and clear, easy to modify later. Nice work!

thanks!

I think if/when we support partial notifications (ie: one account works, one account fails) the Error page will need to morph into an Error/Warning banner.  A future bridge for us to cross

**Comment by setchy** [comment]
Ended up renaming the component back to `Oops` since biome warned about using a global name `Error` (smart!)

File: src/utils/helpers.ts (hunk 2 above)
Comment (afonsojramos) [reply src/utils/helpers.ts:91]: I see 👀 
You are planning on having a central place where you define all error messages aren't you? 👀

**Review by afonsojramos — APPROVED** [review]

File: src/utils/helpers.ts (hunk 2 above)
Comment (setchy) [reply src/utils/helpers.ts:91]: Do you guys want me to remove the try/catch from subject/helper apiRequestAuth calls?

File: .husky/pre-commit (hunk 1 above)
Comment (setchy) [reply .husky/pre-commit:2]: I think i've _finally_ got ontop of my husky dramas.  NVM has been meddling with it

**Pull request merged**